pkg/controllers/rollout/controller.go

```diff
@@ -145,7 +145,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 	}
 
 	// find the master resourceSnapshot.
-	masterResourceSnapshot, err := controller.FetchLatestMasterResourceSnapshot(ctx, r.UncachedReader, placementKey)
+	// Use the cached client so that rollout controller and work-generator have the same view of the
+	// resourceSnapshots in order to reduce the possibility of missing resourceSnapshots in work-generator.
+	masterResourceSnapshot, err := controller.FetchLatestMasterResourceSnapshot(ctx, r.Client, placementKey)
```

Comment:
this actually fixed the unexpected bug which was caused by an uncached read failure but we thought it was from cache.